protocol/app/ante/sigverify.go

```diff
@@ -95,35 +100,24 @@ func (svd SigVerificationDecorator) AnteHandle(
 		// `GoodTilBlock` for replay protection.
 		if !skipSequenceValidation {
 			if accountpluskeeper.IsTimestampNonce(sig.Sequence) {
-				tsNonce := sig.Sequence
-				blockTs := uint64(ctx.BlockTime().UnixMilli())
-				address := acc.GetAddress()
+				start := time.Now()
+				err = svd.akp.ProcessTimestampNonce(ctx, acc, sig.Sequence)
 
-				if !accountpluskeeper.IsValidTimestampNonce(tsNonce, blockTs) {
-					return ctx, errorsmod.Wrapf(
-						sdkerrors.ErrWrongSequence,
-						"timestamp nonce %d not within valid time window", tsNonce,
+				telemetry.MeasureSince(start, []string{metrics.TimestampNonce, metrics.Latency}...)
```

Comment:
Nit: instead of this, add following to start of `ProcessTimestampNonce`:
```
	defer telemetry.ModuleMeasureSince(
		types.ModuleName,
		time.Now(),
		metrics.TimestampNonce,
		metrics.Latency,
	)
	```